[Bug]: Random timeout failures in CI

### Bug summary

We have been struggling with flaky timeout behavior for some time. See https://github.com/matplotlib/matplotlib/issues/29797. At some point we thought, we had fixed it though updating the runner, but it sometimes happens again. So it's not eventually solved. Possibly we have to revive https://github.com/matplotlib/matplotlib/pull/29981 and/or dig deeper.

Failures look like
```
>               raise TimeoutExpired(self.args, orig_timeout)
E               subprocess.TimeoutExpired: Command '['C:\\hostedtoolcache\\windows\\Python\\3.12.10\\x64\\python.exe', '-c', "import importlib.util;_spec = importlib.util.spec_from_file_location('matplotlib.tests.test_backends_interactive', 'D:\\\\a\\\\1\\\\s\\\\lib\\\\matplotlib\\\\tests\\\\test_backends_interactive.py');_module = importlib.util.module_from_spec(_spec);_spec.loader.exec_module(_module);_module._impl_test_lazy_auto_backend_selection()"]' timed out after 20 seconds
```


### Code for reproduction

```Python
-
```


For a start, let's track in which pipelines and for which tests the timeout can be observed:

- Pytest Windows_py311 - test_lazy_auto_backend_selection [ref](https://dev.azure.com/matplotlib/matplotlib/_build/results?buildId=46205&view=logs&j=f92c32cd-721a-536e-3a87-a0c010db3f7f&t=d4eecd58-7131-52e4-f793-e9e3ca32dedb)
- Pytest Windows_py313 - test_interactive_backend[toolbar2-MPLBACKEND=tkagg-BACKEND_DEPS=tkinter] [ref](https://dev.azure.com/matplotlib/matplotlib/_build/results?buildId=46205&view=logs&j=0e51c8a9-4119-5244-b946-ee585b1b50f1&t=e90dd38b-859f-5d2f-90bb-85e3d7473c7a)
- Pytest Windows_py313 - test_fontcache_thread_safe [ref](https://dev.azure.com/matplotlib/matplotlib/_build/results?buildId=46205&view=logs&j=0e51c8a9-4119-5244-b946-ee585b1b50f1&t=e90dd38b-859f-5d2f-90bb-85e3d7473c7a)
- Pytest Windows_py313 - test_fallback_to_different_backend [ref](https://dev.azure.com/matplotlib/matplotlib/_build/results?buildId=46186&j=0e51c8a9-4119-5244-b946-ee585b1b50f1&t=e90dd38b-859f-5d2f-90bb-85e3d7473c7a)
- Pytest Windows_py313 - test_blitting_events[MPLBACKEND=tkagg-BACKEND_DEPS=tkinter] [ref](https://dev.azure.com/matplotlib/matplotlib/_build/results?buildId=46186&j=0e51c8a9-4119-5244-b946-ee585b1b50f1&t=e90dd38b-859f-5d2f-90bb-85e3d7473c7a)
- Pytest Windows_py313 test_blit [ref](https://dev.azure.com/matplotlib/matplotlib/_build/results?buildId=46184&view=logs&j=0e51c8a9-4119-5244-b946-ee585b1b50f1&t=e90dd38b-859f-5d2f-90bb-85e3d7473c7a)
- Pytest Windows_py313 test_fallback_to_different_backend [ref](https://dev.azure.com/matplotlib/matplotlib/_build/results?buildId=46184&view=logs&j=0e51c8a9-4119-5244-b946-ee585b1b50f1&t=e90dd38b-859f-5d2f-90bb-85e3d7473c7a)
- Pytest Windows_py313 test_backend_getattr[matplotlib.backends.backend_agg] [ref](https://dev.azure.com/matplotlib/matplotlib/_build/results?buildId=46184&view=logs&j=0e51c8a9-4119-5244-b946-ee585b1b50f1&t=e90dd38b-859f-5d2f-90bb-85e3d7473c7a)
- Pytest Windows_py312 test_interactive_thread_safety[MPLBACKEND=tkagg-BACKEND_DEPS=tkinter] [ref](https://dev.azure.com/matplotlib/matplotlib/_build/results?buildId=46183&view=logs&j=f92c32cd-721a-536e-3a87-a0c010db3f7f&t=d4eecd58-7131-52e4-f793-e9e3ca32dedb)
- Pytest Windows_py313 test_interactive_timers[MPLBACKEND=tkagg-BACKEND_DEPS=tkinter] [ref](https://dev.azure.com/matplotlib/matplotlib/_build/results?buildId=46171&view=logs&j=0e51c8a9-4119-5244-b946-ee585b1b50f1&t=e90dd38b-859f-5d2f-90bb-85e3d7473c7aj)
- Pytest Windows_py312 test_fontcache_thread_safe [ref](https://dev.azure.com/matplotlib/matplotlib/_build/results?buildId=46169&view=logs&j=f92c32cd-721a-536e-3a87-a0c010db3f7f&t=d4eecd58-7131-52e4-f793-e9e3ca32dedb)
- Pytest Windows_py313 test_blit [ref](https://dev.azure.com/matplotlib/matplotlib/_build/results?buildId=46169&view=logs&j=0e51c8a9-4119-5244-b946-ee585b1b50f1&t=e90dd38b-859f-5d2f-90bb-85e3d7473c7a)
- Pytest Windows_py312 test_qt_missing [ref](https://dev.azure.com/matplotlib/matplotlib/_build/results?buildId=46135&view=logs&j=f92c32cd-721a-536e-3a87-a0c010db3f7f&t=d4eecd58-7131-52e4-f793-e9e3ca32dedb)

### Observations
- seems only to occur on windows, but with all python versions (3.11-3.13)
- if it fails it's typically one test, but up to 4 timed out tests have been observed in a single pytest run
- the Majority of PRs have at least one timeout
- it happens in different tests every time but the common part is that they create subprocesses

--- a/lib/matplotlib/testing/__init__.py
+++ b/lib/matplotlib/testing/__init__.py
@@ -93,7 +93,7 @@ def subprocess_run_for_testing(command, env=None, timeout=60, stdout=None,
     if capture_output:
         stdout = stderr = subprocess.PIPE
     # Add CREATE_NO_WINDOW flag on Windows to prevent console window overhead
-    # This significantly reduces subprocess creation time and helps avoid timeouts
+    # This is added in an attempt to fix flaky timeouts of subprocesses on Windows
     if sys.platform == 'win32':
         if 'creationflags' not in kwargs:
             kwargs['creationflags'] = subprocess.CREATE_NO_WINDOW

